docs/advanced_tutorials/projectors.md

```diff
@@ -0,0 +1,34 @@
+This section introduces the `ProjectorBlock` as an implementation for the the quantum mechanical projection operation onto the subspace spanned by $|a\rangle$: $\mathbb{\hat{P}}=|a\rangle \langle a|$. It evaluates the outer product for bras and kets expressed as bitstrings for a given qubit support. They have to possess matching lengths.
+
+```python exec="on" source="material-block" session="projector" result="json"
+from qadence.blocks import block_to_tensor
+from qadence.operations import Projector  # Projector as an operation.
+
+# Define a projector for |1> onto the qubit labelled 0.
+projector_block = Projector(ket="1", bra="1", qubit_support=0)
+
+# As any block, the matrix representation can be retrieved.
+projector_matrix = block_to_tensor(projector_block)
+
+print(f"projector matrix = {projector_matrix}") # markdown-exec: hide
+```
+
+Other standard operations are expressed as projectors in Qadence. For instance, the number operator $N=\dfrac{1}{2}(I-Z)=|1\rangle\langle 1|$ is used to build projector controlled-unitary gates: $\textrm{CNOT}(i,j)=N(i)\otimes(X(j)-I(j))$.
```

Comment:
Hm do you mean to write here about the generator of controlled-operations? I think that's the only thing you can write with the projector, but you would still need the exponential to make the unitary gate [(e.g. as done here)](https://pasqal-io.github.io/qadence/latest/digital_analog_qc/daqc-cnot/).

Reply:
But I wouldn't necessary include this since we don't really express the controlled unitaries we have as the exponential of the projectors, we mostly just use their direct matrix definitions.

Reply:
Actually, what I really meant is this:

<img width="679" alt="Screenshot 2023-12-18 at 16 03 07" src="https://github.com/pasqal-io/qadence/assets/9250798/99c59f09-4f0f-4e0c-a50a-93951fe93a41">

But this isn't what we really do to define `CNOT`. Maybe a good example is to construct the `CNOT` using projectors and compare the unitaries with our own implementation ?

Reply:
Ah right! Yes sounds good.

Reply:
Done.